blog: Tailscale for DDEV: Simple and Secure Project Sharing (#423)
ddev/ddev.com: src/content/blog/tailscale-router-ddev-addon.md

As a result, you get a stable, human-readable URL for each of your DDEV projects, which you can access from any device on your Tailscale network.

I've found this approach to be particularly useful for:

- **Cross-device testing:** I can easily test my sites on my phone or tablet without being on the same Wi-Fi network.
- **Stable webhook URLs:** I can use the permanent Tailscale URL as a reliable endpoint for webhooks, such as those from payment gateways.
- **Team collaboration:** I can share my development environment with team members so they can see my work in progress.

### How it Works

The `ddev-tailscale-router` add-on works by running a Tailscale container alongside your DDEV project. This container automatically connects to your Tailscale network and securely proxies requests to your project's web container.

### Prerequisites

Before installing the add-on, you need to set up Tailscale:

1. **Install Tailscale** on your development machine by following the [installation guide](https://tailscale.com/download).
2. **Add a second device** to your Tailscale network (phone, tablet, or another computer).

> 3. **Enable HTTPS** in your Tailscale admin console at https://login.tailscale.com/admin/dns by clicking "Enable HTTPS..." This is required for TLS certificate generation.

Reviewer: Let's use the link to the docs here https://tailscale.com/kb/1153/enabling-https

It will be easier to maintain, so you won't have to update this blog if Tailscale decides to move this setting somewhere else.

Author: It is very unlikely that they change this URL. But i have added it too

### Installation

To get started, follow these steps:

1.  First, install the add-on:
    ```bash
    ddev add-on get atj4me/ddev-tailscale-router
    ```

> 2.  Next, get a Tailscale auth key from the [Tailscale admin console](https://login.tailscale.com/admin/settings/keys).

Reviewer: Let's use this docs link here https://tailscale.com/kb/1085/auth-keys

Author: Yep. Changed it!

This add-on uses [Tailscale](https://tailscale.com/), a VPN service that creates a private and secure network between your devices. It is free for personal use!

As a result, you get a stable, human-readable URL for each of your DDEV projects, which you can access from any device on your Tailscale network.

I've found this approach to be particularly useful for:

- **Cross-device testing:** I can easily test my sites on my phone or tablet without being on the same Wi-Fi network.
- **Stable webhook URLs:** I can use the permanent Tailscale URL as a reliable endpoint for webhooks, such as those from payment gateways.
- **Team collaboration:** I can share my development environment with team members so they can see my work in progress.

### How it Works

The `ddev-tailscale-router` add-on works by running a Tailscale container alongside your DDEV project. This container automatically connects to your Tailscale network and securely proxies requests to your project's web container.

### Prerequisites

Before installing the add-on, you need to set up Tailscale:

> 1. **Install Tailscale** on at least one device (phone, tablet, or computer) by following the [installation guide](https://tailscale.com/download). This is needed to access the Tailscale admin console and generate an auth key.

Reviewer: First time setup on two devices (not one) is a requirement, otherwise auth key is not available at all.

Author: Oh. Didn't realize that. It woked with a single device when I started last year. Anyways, has updated the section

    ```

2.  Next, **install the add-on:**

    ```bash
    ddev add-on get atj4me/ddev-tailscale-router
    ```

3.  Finally, **restart DDEV:**
    ```bash
    ddev restart
    ```

### Using Your Tailscale URL

Once installation is complete, you can access your project using these commands:

Launch your project's Tailscale URL in browser:

```bash
ddev tailscale launch
```

Get your project's Tailscale URL:

```bash
ddev tailscale url
```

Your project's permanent Tailscale URL will look like: `https://<project-name>.<your-tailnet>.ts.net`. You can also find it in your Tailscale admin console.

### Public vs. Private Mode

The add-on offers two modes for sharing your project:

- **Private (default):** Your project is only accessible to devices on your Tailscale network.
- **Public:** Your project is accessible to anyone on the internet.

To switch between modes:

Switch to public mode (accessible to anyone on the internet):

```bash
ddev dotenv set .ddev/.env.tailscale-router --ts-privacy=public
ddev restart
```

Switch back to private mode (default):

```bash
ddev dotenv set .ddev/.env.tailscale-router --ts-privacy=private
ddev restart
```

> > **Note:** For public access, you need to configure your [Access Control List (ACL)](https://tailscale.com/kb/1223/funnel#funnel-node-attribute) to enable Funnel. See the [Tailscale Funnel documentation](https://tailscale.com/kb/1223/funnel) for details on setting up the required ACL policy.

Author: They added that UI very recently. It wasn't there in July. I guess they still haven't updated the Readme, since this is still in Beta

---

> title: "A Simple and Secure Way to Share Your DDEV Projects with Tailscale"

Reviewer: Does it make sense to rephrase it this way, placing the main subject at the start?:

```suggestion
title: "Tailscale for DDEV: Simple and Secure Project Sharing"
```

pubDate: 2025-08-26
summary: "Tired of temporary sharing links? Learn how to use the ddev-tailscale-router add-on to get a stable, secure, and private URL for your DDEV projects."
author: Ajith Thampi Joseph
featureImage:
  src: /img/blog/2025/08/ddev_tailscale_flowchart.svg
  alt: "DDEV Tailscale Router Cover Image"
  caption: "The ddev-tailscale-router add-on enables cross-device testing, stable webhook URLs, and team collaboration."
  credit: "Image by Claude AI :)"
categories:
  - Guides
  - TechNotes
---

> I've found that DDEV's `ddev share` command is a great way to quickly share my local development environment. However, since it uses ngrok, it generates a new, random URL every time, which can be a hassle. For a more permanent and secure solution, I've created the `ddev-tailscale-router` add-on.

Reviewer: I would make a link to the add-on in the first paragraph so that people can go there right away.

> This add-on uses Tailscale, a service that creates a private and secure network between your devices. As a result, you get a stable, human-readable URL for each of your DDEV projects, which you can access from any device on your Tailscale network.

Reviewer: I think this also needs a link to the Tailscale website.

And Tailscale is not free, is it? Then we need a mention for that.

---
title: "Tailscale for DDEV: Simple and Secure Project Sharing"
pubDate: 2025-08-26
summary: "Tired of temporary sharing links? Learn how to use the ddev-tailscale-router add-on to get a stable, secure, and private URL for your DDEV projects."
author: Ajith Thampi Joseph
featureImage:
  src: /img/blog/2025/08/ddev_tailscale_flowchart.svg
  alt: "DDEV Tailscale Router Cover Image"
  caption: "The ddev-tailscale-router add-on enables cross-device testing, stable webhook URLs, and team collaboration."
  credit: "Image by Claude AI :)"
categories:
  - Guides
  - TechNotes
---

> I've found that DDEV's [`ddev share`](https://docs.ddev.com/en/stable/users/topics/sharing/) command is a great way to quickly share my local development environment. However, since it uses ngrok, it generates a new, random URL every time, which can be a hassle. For a more permanent and secure solution, I've created the [`ddev-tailscale-router`](https://github.com/atj4me/ddev-tailscale-router) add-on.

Reviewer 2: Ngrok can easily be used with a stable URL, and the technique is covered in the docs. That's not the only reason you created this is it?

```suggestion
I've found that DDEV's [`ddev share`](https://docs.ddev.com/en/stable/users/topics/sharing/) command is a great way to quickly share my local development environment. However, since it uses ngrok, it generates a new, random URL every time unless you use a [stable domain](https://docs.ddev.com/en/stable/users/topics/sharing/#setting-up-a-stable-ngrok-domain). As an alternative, I've created the [`ddev-tailscale-router`](https://github.com/atj4me/ddev-tailscale-router) add-on.
```

Reviewer 2: You only get one stable domain for free :)

Author: Yeah, that and the option to access it on my mobile without putting the entire thing on the internet

> This add-on uses [Tailscale](https://tailscale.com/), a service that creates a private and secure network between your devices. It is free for personal use!

Reviewer 2: ```suggestion
This add-on uses [Tailscale](https://tailscale.com/), a VPN service that creates a private and secure network between your devices. It is free for personal use!
```

As a result, you get a stable, human-readable URL for each of your DDEV projects, which you can access from any device on your Tailscale network.

I've found this approach to be particularly useful for:

- **Cross-device testing:** I can easily test my sites on my phone or tablet without being on the same Wi-Fi network.
- **Stable webhook URLs:** I can use the permanent Tailscale URL as a reliable endpoint for webhooks, such as those from payment gateways.
- **Team collaboration:** I can share my development environment with team members so they can see my work in progress.

### How it Works

The `ddev-tailscale-router` add-on works by running a Tailscale container alongside your DDEV project. This container automatically connects to your Tailscale network and securely proxies requests to your project's web container.

### Installation

> To get started, follow these steps:

Reviewer: It makes sense to add a warning about Apple Silicone from https://github.com/atj4me/ddev-tailscale-router#getting-started

We have the majority of users on macOS https://ddev.com/blog/stats-on-ddev-usage-nov-2024/#ddevenvironments

It's important to remember where it can be used.

1.  First, install the add-on:
    ```bash
    ddev add-on get atj4me/ddev-tailscale-router
    ```

> 2.  Next, get a Tailscale auth key from the [Tailscale admin console](https://login.tailscale.com/admin/settings/keys).

Reviewer 2: Yes, I was logged in. 
<img width="1282" height="787" alt="image" src="https://github.com/user-attachments/assets/d01a6769-7811-44ed-9e0a-95bc51372fef" />

> 2.  Next, get a Tailscale auth key from the [Tailscale admin console](https://login.tailscale.com/admin/settings/keys).

Author: I guess you need to complete it.
